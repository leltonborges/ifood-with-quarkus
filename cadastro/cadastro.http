
### Get todos os restaurantes
GET localhost:8090/restaurantes
Accept: application/json