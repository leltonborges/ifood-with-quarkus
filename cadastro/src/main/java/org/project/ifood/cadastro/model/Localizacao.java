package org.project.ifood.cadastro.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;

@Entity
@Table(name = "tb_localizacao")
public class Localizacao extends PanacheEntityBase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public Double latitude;
    public Double longitude;
}