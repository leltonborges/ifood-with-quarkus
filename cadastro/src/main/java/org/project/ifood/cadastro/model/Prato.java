package org.project.ifood.cadastro.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "tb_prato")
public class Prato extends PanacheEntityBase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String nome;
    public String descricao;

    @ManyToOne
    public Restaurante restaurante;
    public BigDecimal preco;
}
