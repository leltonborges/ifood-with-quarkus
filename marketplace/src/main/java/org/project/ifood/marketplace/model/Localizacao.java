package org.project.ifood.marketplace.model;

import io.quarkus.hibernate.reactive.panache.PanacheEntityBase;

import javax.persistence.*;

@Entity
@Table(name = "tb_localizacao")
public class Localizacao extends PanacheEntityBase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public Double latitude;
    public Double longitude;
}
