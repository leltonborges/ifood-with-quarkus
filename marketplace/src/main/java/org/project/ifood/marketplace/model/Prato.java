package org.project.ifood.marketplace.model;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
public class Prato {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String nome;
    public String descricao;

    @ManyToOne
    public Restaurante restaurante;
    public BigDecimal preco;
}
