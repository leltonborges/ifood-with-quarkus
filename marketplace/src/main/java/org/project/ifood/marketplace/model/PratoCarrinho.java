package org.project.ifood.marketplace.model;

import io.quarkus.hibernate.reactive.panache.PanacheEntityBase;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "tb_prato_cliente")
public class PratoCarrinho extends PanacheEntityBase {
    @Id
    public Long id;
    public String usuario;

}
