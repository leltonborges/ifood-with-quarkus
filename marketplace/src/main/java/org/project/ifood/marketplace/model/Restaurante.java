package org.project.ifood.marketplace.model;

import io.quarkus.hibernate.reactive.panache.PanacheEntityBase;

import javax.persistence.*;

@Entity
@Table(name = "tb_restaurante")
public class Restaurante extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    public String nome;
    @OneToOne(cascade = CascadeType.ALL)
    public Localizacao localizacao;
}
