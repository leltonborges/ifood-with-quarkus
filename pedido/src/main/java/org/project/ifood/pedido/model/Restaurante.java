package org.project.ifood.pedido.model;

public class Restaurante {
    public Long idRestaurante;
    public String cnpj;
    public String nome;
}
