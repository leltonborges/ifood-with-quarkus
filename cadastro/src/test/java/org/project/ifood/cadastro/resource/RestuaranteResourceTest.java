package org.project.ifood.cadastro.resource;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;
import org.project.ifood.cadastro.CadastroTestLifecycleManager;
import org.project.ifood.cadastro.model.Restaurante;

import javax.ws.rs.core.Response;

import static io.restassured.RestAssured.given;
import static io.restassured.RestAssured.when;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestuaranteResourceTest {

    @Test
    @DataSet("restaurante-cenario-1.yaml")
    public void testBuscandoRestaurantes(){
        String result = when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();

        Approvals.verify(result);
    }

    @Test
    @DataSet("restaurante-cenario-1.yaml")
    public void testAlterarRestaurante(){
        Restaurante dto = new Restaurante();
        dto.nome = "novoNome";
        Long parametroValor = 123L;
        given()
                .with().pathParam("id", parametroValor)
                .body(dto)
                .when()
                .put("/restaurantes/{id}")
                .then()
                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
                .extract().asString();
        Restaurante entidade = Restaurante.findById(parametroValor);
        Assert.assertEquals(dto.nome, entidade.nome);
    }

    private RequestSpecification given(){
        return RestAssured.given().contentType(ContentType.JSON);
    }
}
