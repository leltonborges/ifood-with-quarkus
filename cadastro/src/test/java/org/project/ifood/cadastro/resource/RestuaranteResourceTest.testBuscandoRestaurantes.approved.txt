[{"id":123,"nomeFantasia":"Novo restauraten","proprietario":"ID do keyclock 123"},{"id":234,"nomeFantasia":"Novo restauratene","proprietario":"ID do keyclock 234"}]